services/workflows-service/src/customer/schemas/zod-schemas.ts

```diff
@@ -0,0 +1,19 @@
+import { z } from 'zod';
+
+export const SubscriptionSchema = z.discriminatedUnion('type', [
```

Comment:
We should probably reuse our existing schema and just extract it to a common place